lib/src/slices/user_referral/user_referral_service.dart

```diff
@@ -17,42 +19,56 @@ class UserReferralService extends ChangeNotifier {
   late final UserReferralPresenter presenter;
   late final UserReferralController controller;
   late final UserReferralModel model;
+  final ApiAppDataService apiAppDataService;
+  final LoginFlowService loginFlowService;
+  final ApiSignupService apiSignupService;
 
-  UserReferralService() {
+  UserReferralService(
+      this.apiAppDataService, this.loginFlowService, this.apiSignupService) {
     this.presenter = UserReferralPresenter(this);
     this.controller = UserReferralController(this);
     this.model = UserReferralModel();
+    getCode();
   }
 
-  String getCode(LoginFlowService loginFlowService) {
-    String? code = loginFlowService.model.user!.user!.code;
-    if (code == null || code.isEmpty) _updateCode(loginFlowService);
+  String getCode() {
+    String? code = this.model.code;
+    if (code == null || code.isEmpty)
+      _updateCode(this.apiAppDataService, this.loginFlowService)
+          .then((_) => updateReferCount());
```

Comment:
this works, but if you wanted you could also just make it an async func with an await